### PR TITLE
Add pre/post script support

### PR DESCRIPTION
**Pull request by rstyd**
This PR addresses #773. It adds pre/post script support through a `beeflow:ScriptRequirement` hint which has the format:

```
    beeflow:ScriptRequirement:
    enabled: true
    pre_script_path: "pre_run.sh"
    post_script_path: "post_run.sh
```

The pre_run and post_run scripts are read using the parser and passed to the task object for the associated task. It's then read and added to the SBATCH script in the slurm runner.

### CODE REVIEW

**Comment by pagrubel** [comment]
@rstyd Your example works. Please add it to all the workers and can you set up a test in CI? I think there should be a way to att it to the Flux and Slurm tests using a simpler example.

**Review by pagrubel — CHANGES_REQUESTED** [review]
As I commented it should be added to all the workers and if possible make CI tests.

**Comment by pagrubel** [comment]
1. Add pre/post to all workers.
2. Add test to slurm and flux integration test.
3. Documentation.

**Review by kchilleri — DISMISSED** [review]
Same pre/post script support will be added to the flux_worker (as requested by Pat) in a separate issue/PR. 
Test/documentation TBD?

**Comment by pagrubel** [comment]
@kchilleri There may be some misunderstanding. Yes the flux implementation should be separate. However, this PR is still missing documentation and integration tests. So really it shouldn't be approved as is.

**Comment by kchilleri** [comment]
@pagrubel Got it! Should I dismiss my approval?

**Review by kchilleri — DISMISSED** [review]

**Review by pagrubel — APPROVED** [review]

**Review by pagrubel — CHANGES_REQUESTED** [review]
Thanks, @rstyd All looks good. Please squash the commits and let me know when you take out your debug Asserts.

File: docs/sphinx/bee_cwl.rst
Comment (pagrubel) [inline docs/sphinx/bee_cwl.rst]:
```suggestion
and after a task. The script must be within the workflow directory. The shell
```

**Review by pagrubel — CHANGES_REQUESTED** [review]

**Review by pagrubel — APPROVED** [review]

**Comment by pagrubel** [comment]
@rstyd Please squash the commits, or I can if you wish.

**Pull request merged**
